### DISCUSSION TITLE
Documentation for uart interrupts

### DISCUSSION THREAD

**Discussion by maciekish** [stm32duino/Arduino_Core_STM32#1975] [Q&A]
Hi,
I need to port some uart code that uses interrupts from the old libmaple core, but I cannot find any documentation regarding uart and interrupts in this core. Where do I start?

Thanks
Maciej

**Comment by fpistm** [comment]
Uart IT is used behind the scene. You could have look to HardwareSerial files.

https://github.com/stm32duino/Arduino_Core_STM32/blob/4fab426eab45f0156330779a46470ce92e17f415/cores/arduino/HardwareSerial.cpp#L332-L373
https://github.com/stm32duino/Arduino_Core_STM32/blob/4fab426eab45f0156330779a46470ce92e17f415/libraries/SrcWrapper/src/stm32/uart.c#L801-L844
